client/services/SolutionHandlerServiceSpec.js

```diff
@@ -0,0 +1,240 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for SolutionHandlerService.
+ */
+
+describe('SolutionHandlerService', function() {
+  var SolutionHandlerService;
+  var TaskObjectFactory;
+  var orderedTasks;
+  var auxiliaryCode;
+  var startCode;
+
+  beforeEach(module('tie'));
+  beforeEach(inject(function($injector) {
+    SolutionHandlerService = $injector.get(
+      'SolutionHandlerService');
+  }));
+
+  // Mock tasks for preprocessing.
+  beforeEach(inject(function($injector) {
+    TaskObjectFactory = $injector.get(
+      'TaskObjectFactory');
+
+    var taskDict = [
+      {
+        instructions: [''],
+        prerequisiteSkills: [''],
+        acquiredSkills: [''],
+        inputFunctionName: null,
+        outputFunctionName: null,
+        mainFunctionName: 'mockMainFunction',
+        correctnessTests: [{
+          input: 'task_1_correctness_test_1',
+          allowedOutputs: [true]
+        }, {
+          input: 'task_1_correctness_test_2',
+          allowedOutputs: [true]
+        }],
+        buggyOutputTests: [{
+          buggyFunctionName: 'AuxiliaryCode.mockAuxiliaryCodeOne',
+          messages: [
+            "Mock BuggyOutputTest Message One for task1",
+            "Mock BuggyOutputTest Message Two for task1",
+            "Mock BuggyOutputTest Message Three for task1"
+          ]
+        }],
+        performanceTests: []
+      },
+      {
+        instructions: [''],
+        prerequisiteSkills: [''],
+        acquiredSkills: [''],
+        inputFunctionName: null,
+        outputFunctionName: null,
+        mainFunctionName: 'mockMainFunction',
+        correctnessTests: [{
+          input: 'task_2_correctness_test_1',
+          allowedOutputs: [false]
+        }, {
+          input: 'task_2_correctness_test_2',
+          allowedOutputs: [false]
+        }],
+        buggyOutputTests: [{
+          buggyFunctionName: 'AuxiliaryCode.mockAuxiliaryCodeTwo',
+          messages: [
+            "Mock BuggyOutputTest Message One for task2",
+            "Mock BuggyOutputTest Message Two for task2",
+            "Mock BuggyOutputTest Message Three for task2"
+          ]
+        }],
+        performanceTests: []
+      }
+    ];
+
+    orderedTasks = taskDict.map(function(task) {
+      return TaskObjectFactory.create(task);
+    });
+  }));
+
+  // Mock auxiliary code for processing.
+  beforeEach(function() {
```

Comment:
Can you include the stuff in this and the next beforeEach clause into the above one (i.e. after defining orderedTasks in line 88)? We don't need separate beforeEach clauses for each variable assignment.